cvat/apps/engine/views.py

```diff
@@ -874,6 +874,29 @@ def annotations(self, request, pk):
                     return Response(data=str(e), status=status.HTTP_400_BAD_REQUEST)
                 return Response(data)
 
+    @action(detail=True, methods=['POST'], url_path=r'(?P<version>\d+)/assign/(?P<user_id>\d+)')
+    def assign(self, request, pk, version, user_id):
+        version = int(version)
+        user_id = int(user_id)
+        if user_id == 0:
+            user_id = None
+
+        with transaction.atomic():
+            segment = models.Segment.objects.select_for_update().get(job__id=pk)
+            if segment.concurrent_version != version:
+                return Response(status=status.HTTP_409_CONFLICT)
+
+            current_job, other_jobs = segment.split_jobs(pk)
+            if user_id and any(j.assignee_id == user_id for j in other_jobs):
+                raise serializers.ValidationError("User is already working on another job for the given sequence.")
```

Comment:
why returning error response above but throwing exception here?
why difference in error handling?

Reply:
Raising `ValidationError` causes server to return 400 bad request response, which says that submitted data is invalid, but when concurrent update happens and error response is sent, the submitted data is valid. So IMO it's inappropriate to send 400 response in this case.